model_compression_toolkit/core/keras/pruning/pruning_keras_implementation.py

```diff
@@ -0,0 +1,117 @@
+from typing import List, Tuple
+
+from model_compression_toolkit.core.common.framework_info import FrameworkInfo
+from model_compression_toolkit.core.common import BaseNode
+from model_compression_toolkit.core.common.pruning.pruning_framework_implementation import \
+    PruningFrameworkImplementation
+from model_compression_toolkit.core.keras.keras_implementation import KerasImplementation
+from model_compression_toolkit.core.keras.pruning.attributes_info import get_keras_node_attributes_with_io_axis
+from model_compression_toolkit.core.keras.pruning.check_node_role import is_keras_node_intermediate_pruning_section, \
+    is_keras_entry_node, is_keras_exit_node
+from model_compression_toolkit.core.keras.pruning.prune_keras_node import (prune_keras_exit_node,
+                                                                           prune_keras_entry_node, \
+                                                                           prune_keras_intermediate_node)
+import numpy as np
+
+class PruningKerasImplementation(KerasImplementation, PruningFrameworkImplementation):
+    """
+    Implementation of the PruningFramework for the Keras framework. This class provides
+    concrete implementations of the abstract methods defined in PruningFrameworkImplementation
+    for the Keras framework.
+    """
+
+    def prune_entry_node(self, node: BaseNode, output_mask: np.ndarray, fw_info: FrameworkInfo):
+        """
+        Prunes the entry node of a model in Keras.
+
+        Args:
+            node: The entry node to be pruned.
+            output_mask: A numpy array representing the mask to be applied to the output channels.
+            fw_info: Framework-specific information object.
+
+        Returns:
+            The result from the pruning operation.
+        """
+        return prune_keras_entry_node(node, output_mask, fw_info)
```

Comment:
connected to the previous comment - why isn't the implementation here? (same for the rest of the methods)